### DISCUSSION TITLE
Requirements for auto generated USDZ?

### DISCUSSION THREAD

**Discussion by DominikDa** [google/model-viewer#3500] [Q&A]
Hi,

which are the exact requirements for the auto generation of USDZs on the fly while manipulating the model through the API?

For my model it doesn’t work, on iOS an error shows that says „To show this model you need to update your iOS version“.

I am not declaring an ios-src and I set ar-modes to webxr scene-viewer quick-look.

Is there any other requirement for the auto generation to work?

Thanks!

**Comment by samaneh-kazemi** [comment]
Hi @DominikDa, thanks for reaching out. That's supported for [supported iOS 12+ devices](https://www.apple.com/ios/augmented-reality/). If that doesn't help, please remix our https://glitch.com/edit/#!/model-viewer with your exact settings so we can look into it.

**Reply by DominikDa** [reply]
Hi @samaneh-kazemi, thanks for getting back to me. We have tested with the newest iOS 15 iPhones and we cannot get the model to work.

I have reduced the settings as much as possible and came up with this example: https://peridot-silk-snowshoe.glitch.me/
Note that in general the iOS devices are able to display the object in AR. Only after playing with the color input it does not work anymore.

Thanks for looking into this!

**Reply by elalish** [reply]
USDZ auto-generation doesn't work for all models, as many features are still missing. Still, this sounds like it could likely be a USDZExporter bug in three. Your Glitch doesn't seem to be public; can you make it accessible?

**Reply by DominikDa** [reply]
Thanks for looking into this, here's a new Glitch https://guttural-mixed-honey.glitch.me/

**Reply by elalish** [reply]
Interesting, it exports and opens Quick Look, but then Quick Look just hangs forever with the spinner (or at least longer than I felt like waiting). @grorg, can you repro? Even if the USDZ is a little weird, iOS should probably be able to give some more useful information.
